File: examples/google-adk-python/hot_path.py
import asyncio
import os

from memexai import MemexAI


async def main() -> None:
    memex = MemexAI(url=os.getenv("MEMEX_URL", "http://localhost:8080"), api_key=os.getenv("MEMEX_API_KEY", "dev-agent-key"))
    try:
        memory = memex.for_user(os.getenv("MEMEX_USER_ID", "adk_hot_path_user"), actor="google-adk-hot-path")
        result = await memory.remember("Remember that ADK hot-path memory is for explicit current-turn saves.")
        print("hot_path: call memory.remember from an ADK action/tool when durability is part of the current turn")
        print(result.get("writes", []))
    finally:
        await memex.close()


if __name__ == "__main__":
    asyncio.run(main())

File: examples/anthropic-service/src/hot-path.ts
import { MemexAI } from "@memexai/sdk"

const memory = new MemexAI({
  url: process.env["MEMEX_URL"] ?? "http://localhost:8080",
  apiKey: process.env["MEMEX_API_KEY"] ?? "dev-agent-key",
}).forUser({
  userId: process.env["MEMEX_USER_ID"] ?? "anthropic_service_hot_path_user",
  actor: "anthropic-service-hot-path",
})

const result = await memory.remember({
  text: "Remember that I prefer Anthropic tool-use loops with explicit tool_result blocks.",
})

console.log("hot_path: awaited memory.remember before finishing the turn")
console.log(`writes: ${result.writes.map((write) => write.path).join(", ") || "none"}`)
